### PR TITLE
Add new parameter to somerc projection

### PR DESCRIPTION
**Pull request by istvanhajdu**
**Feature description**
The somerc projection uses a double projection from ellipsoid through a conformal sphere to cylinder. In some unique cases (in the Hungarian national projection for instance) the latitude of the projection centre may differ in the two projection steps ([see details](https://ufile.io/h17p8zfz)). A new optional parameter (+latc=) was added to precisely handle those cases.

**Problem description**
After the implementation of the new parameter we have noticed that the cs2cs  provided​ wrong result although with the pipline operation the result was the expected.

**Reproduction of the problem:**

*With cs2cs:*

`echo 20.0 47.0 100.0 | cs2cs  -f "%.4f" +init=epsg:4258 +to +proj=somerc +lat_0=47.1666666666667 +lon_0=19.04857177777778 +latc=47.1 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67  +nadgrids=etrs2eov_notowgs.gsb +geoidgrids=geoid_eht2014.gtx +units=m`
`722440.3617     184418.8430 14.9214`

*With pipline:*

`
echo 20.0 47.0 100.0 2022.0 | cct +proj=pipeline +step +proj=vgridshift +grids=geoid_eht2014.gtx +step +proj=hgridshift  +grids=etrs2eov_notowgs.gsb +inv +step +proj=somerc +lat_0=47.1666666666667 +lon_0=19.04857177777778 +latc=47.1 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +units=m
`
`722440.3617    184418.8430       57.4600     2022.0000`

**Grid files:**
- [etrs2eov_notowgs.gsb​](https://ufile.io/xcf9y9p6)
- [geoid_eht2014.gtx](https://ufile.io/dtr69eaa)

**Environment**

- Proj 8.2.0
- Debian 11
- Build with CMake

### CODE REVIEW

**Comment by rouault** [comment]
> After the implementation of the new parameter we have noticed that the cs2cs provided​ wrong result although with the pipline operation the result was the expected.

This is very likely due to https://github.com/OSGeo/PROJ/blob/021d6bbabf7eed684db0dbee3a53e9a34c2b19dc/src/iso19111/io.cpp#L10000 that maps somerc to HOTINE_OBLIQUE_MERCATOR_VARIANT_B and thus discard your additional parameter. You'd likely have to add a new projection method and mapping in src/iso19111/operation/parammappings.cpp

Did you reach with IOGP / EPSG about this variant of hotine oblique mercator ? (https://epsg.org/dataset-change-requests.html). It'd be better if EPSG acknowledged that and allocated, let's say, a HOTINE_OBLIQUE_MERCATOR_VARIANT_C method for that situation.

File: docs/source/operations/projections/somerc.rst [hunk 1]
```diff
@@ -28,3 +28,7 @@ Parameters
 .. include:: ../options/x_0.rst
 
 .. include:: ../options/y_0.rst
+
+.. option:: +latc=<value>
+
+   Normal latitude of the ellipsoid
```
Comment (rouault) [inline docs/source/operations/projections/somerc.rst:34]: A bit more explanation would be needed, like an extract from the summary of the paper + a link to it.

**Comment by kbevers** [comment]
@istvanhajdu if you can make the changes requested by @rouault be tomorrow afternoon CET this will be included in PROJ 9.0.0. Next change is in June.

**Comment by istvanhajdu** [comment]
@rouault @kbevers Currently we are working on the EPSG request. If the committee approve the request I will indicate it and come back to make the appropriate code changes in this PR.

**Pull request closed without merging**